src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -1424,9 +1425,20 @@ private static void validateSSTableFormatFactories(Iterable<SSTableFormat.Factor
         if (selectedFormat == null)
             throw new ConfigurationException(String.format("Selected sstable format '%s' is not available.", selectedFormatName));
 
+        validateWriteFormatVsStorageCompatibilityMode(selectedFormat, getStorageCompatibilityMode());
+
         return selectedFormat;
     }
 
+    @VisibleForTesting
+    public static void validateWriteFormatVsStorageCompatibilityMode(SSTableFormat<?, ?> selectedFormat, StorageCompatibilityMode mode)
+    {
+        if (selectedFormat.name().equals(BtiFormat.NAME) && mode != StorageCompatibilityMode.NONE)
```

Comment:
To be on the safe side and not muddy the waters further while on mixed compatibility mode clusters. That would just complicate things further for no/little benefit imo.

Reply:
I really dislike having moving parts while `UPGRADING`. This means we'd have to test the whole codebase in `UPGRADING` which is yet another CI duplication of work. But tries might be a special case being behind a pretty solid feature flag. Also avoiding the extra oa hop while upgrading seems worth it. Lets do it for tries.

Reply:
That situation, being able to support new features during rolling upgrades, is not new. I don't think that offering the possibility of running in compatibility mode changes it. IMO it doesn't remove the need of having the rolling upgrade dtests that we usually add to prevent multiplexing entire CI. For the scope of this ticket, I think we can assume that tries work in rolling upgrade mode, since that was the assumption when they were merged.